homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,579 @@
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, INPUT_NUMBER_DOMAIN, NUMBER_DOMAIN]
+            )
+        ),
+        vol.Optional(CONF_ABOVE): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+        vol.Optional(CONF_BELOW): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+    },
+).extend(SUBSCHEMA_BOILERPLATE.schema)
+
+TEMPLATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_VALUE_TEMPLATE): selector.TemplateSelector(
+            selector.TemplateSelectorConfig(),
+        ),
+    },
+).extend(SUBSCHEMA_BOILERPLATE.schema)
+
+
+class ConfigFlowSteps(StrEnum):
+    """StrEnum for all the different config steps."""
+
+    USER = "user"
+    OBSERVATION_SELECTOR = "observation_selector"
+
+
+class OptionsFlowSteps(StrEnum):
+    """StrEnum for all the different options flow steps."""
+
+    INIT = "init"
+    BASE_OPTIONS = "base_options"
+    ADD_OBSERVATION = str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+    SELECT_EDIT_OBSERVATION = "select_edit_observation"
+    EDIT_OBSERVATION = "edit_observation"
+    REMOVE_OBSERVATION = "remove_observation"
+
+    @staticmethod
+    def list_primary_steps() -> list[str]:
+        """Return a list of the values."""
+        li = [c.value for c in OptionsFlowSteps]
+        li.remove("init")
+        li.remove("edit_observation")
+        return li
+
+
+def _convert_percentages_to_fractions(
+    data: dict[str, str | float | int],
+) -> dict[str, str | float]:
+    """Convert percentage values in a dictionary to fractions."""
+    percentages = [
+        CONF_P_GIVEN_T,
+        CONF_P_GIVEN_F,
+        CONF_PRIOR,
+        CONF_PROBABILITY_THRESHOLD,
+    ]
+    return {
+        key: (
+            value / 100
+            if isinstance(value, (int, float)) and key in percentages
+            else value
+        )
+        for key, value in data.items()
+    }
+
+
+def _convert_fractions_to_percentages(
+    data: dict[str, str | float],
+) -> dict[str, str | float]:
+    """Convert fraction values in a dictionary to percentages."""
+    percentages = [
+        CONF_P_GIVEN_T,
+        CONF_P_GIVEN_F,
+        CONF_PRIOR,
+        CONF_PROBABILITY_THRESHOLD,
+    ]
+    return {
+        key: (
+            value * 100
+            if isinstance(value, (int, float)) and key in percentages
+            else value
+        )
+        for key, value in data.items()
+    }
+
+
+async def _get_select_observation_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Return schema for selecting a observation."""
+    return vol.Schema(
+        {
+            vol.Required(CONF_INDEX): vol.In(
+                {
+                    str(index): f"{config.get(CONF_NAME)} ({config[CONF_PLATFORM]})"
```

Comment:
Is there a way to localise this string but still keep the dynamic contents. And also to give a friendly/localised names to `{config[CONF_PLATFORM]}` which is otherwise a string literal? Same with L277

Reply:
There is only one platform in this helper so why we're using it for anything seems strange to me?
Also that each observation has a name feels strange?

Reply:
As discussed elsewhere "PLATFORM" in this context means one of "numeric_state", "to_state" or "template". Which are the three different forms of "observations".
Names are required so that observations can be selected for deleting or editing, as there is no simpler way to uniquely identify each observation in a user-friendly way (template observations are especially hard)